### PR TITLE
remove default template loading on exception

### PR DESCRIPTION
**Pull request by greg0ire**
Why would anyone want to rely on that? Maybe there is a valid reason,
but it does not appear in the comments or in the commit history. The
best way to find out is to remove the "feature". If anyone reverts this
commit, please make sure to add a comment to explain this.

### CODE REVIEW

**Comment by greg0ire** [comment]
I have run my project's test suite with this piece of code, and it passed. I'll keep it in my project for a while, but I think this really can be removed.

**Comment by greg0ire** [comment]
This is this kind of BC-break : 

xkcd

**Comment by pulzarraider** [comment]
Refs  #3255. 

Even this is BC break, :+1:. Debugging this kind of typo in template can be very paintfull.
To not create BC break, the existing code should be kept, but some exception should occurs in dev mode.

**Comment by greg0ire** [comment]
> Debugging this kind of typo in template can be very paintfull.
To not create BC break, the existing code should be kept, but some exception should occurs in dev mode.

Yes, but I don't really want to keep BC.

**Comment by Soullivaneuh** [comment]
Please fill your PR of missing parts indicated by the "Needs" labels.

Feel free to ask if you want some precision of what is needed.

**Comment by greg0ire** [comment]
@Soullivaneuh done

**Comment by greg0ire** [comment]
@sonata-project/contributors final review ?

**Comment by Soullivaneuh** [comment]
Personally, I would like a BC way with a `trigger_error`.

If we can follow the Symfony method (new major is the last minor without deprecated stuff), it would be wonderful. :+1: 

In any case, an upgrade note is needed.

**Comment by greg0ire** [comment]
Ok, I'll do that.

**Comment by greg0ire** [comment]
Temporarily closing so that this does not get merge by error.

File: Tests/Twig/Extension/SonataAdminExtensionTest.php [hunk 1]
```diff
@@ -1144,57 +1078,13 @@ class="x-editable"
...
-            ))));
-
-        $this->twigExtension->renderListElement($this->environment, $this->object, $this->fieldDescription);
-    }
 
     /**
      * @expectedException        Twig_Error_Loader
-     * @expectedExceptionMessage Unable to find template "base_list_nonexistent_field.html.twig"
-     * @group                    legacy
+     * @expectedExceptionMessage Unable to find template "list_nonexistent_template.html.twig"
```
Comment (greg0ire) [inline Tests/Twig/Extension/SonataAdminExtensionTest.php]: Not sure whether this test should be kept… it depends on Twig.

**Comment by greg0ire** [comment]
ping @sonata-project/contributors

**Pull request merged**